### PR TITLE
Add option for `outbuffer` to indent by 4 spaces

### PR DESCRIPTION
**Pull request by UplinkCoder**
... and document indentation feature.

### CODE REVIEW

**Comment by Geod24** [comment]
Maybe make it an `enum` ?

```D
enum Indent
{
    None,
    Spaces,
    Tab,
}
```

**Comment by UplinkCoder** [comment]
@Geod24 Nope. I want thie api change to be additive only.

**Comment by UplinkCoder** [comment]
Does anyone have any insight in why the doctest fails?

**Comment by UplinkCoder** [comment]
The tests on the major platforms pass  @Geod24  (of course they do nothing changed)

File: src/dmd/root/outbuffer.d [hunk 1]
```diff
@@ -151,7 +157,6 @@ struct OutBuffer
         offset += len;
     }
 
-    // write newline
```
Comment (Geod24) [inline src/dmd/root/outbuffer.d]: Unrelated?

**Review by Geod24 — APPROVED** [review]
Just a nit about  unrelated comment removal, otherwise LGTM

**Comment by Geod24** [comment]
Master is fixed so next rebase / PR won't have the issue.

**Comment by UplinkCoder** [comment]
@Geod24 fixed ready for merge.

**Comment by dkorpel** [comment]
This could use a unittest

File: src/dmd/root/outbuffer.d [hunk 2]
```diff
@@ -28,7 +28,12 @@ struct OutBuffer
     private ubyte[] data;
     private size_t offset;
     private bool notlinehead;
+
+    /// Whether to indent
     bool doindent;
+    /// Whether to indent by 4 spaces or by tabs;
+    bool spaces;
+    /// Current indent levlel
```
Comment (dkorpel) [inline src/dmd/root/outbuffer.d]: level

Comment (UplinkCoder) [reply src/dmd/root/outbuffer.d]: why this is extremely unlikely to ever break.
And it's going to be tested in dtoh the tests.
Therefore a unit-test is just a waste to everyone's time if they have to wait for the auto-tester to finish.

File: src/dmd/root/outbuffer.d (hunk 1 above)
Comment (UplinkCoder) [reply src/dmd/root/outbuffer.d]: it has to do with documenting the indent ability.
which won't work if you don't use `writenl`

**Review by Geod24 — APPROVED** [review]

File: src/dmd/root/outbuffer.d (hunk 2 above)
Comment (Geod24) [reply src/dmd/root/outbuffer.d]: > Therefore a unit-test is just a waste to everyone's time if they have to wait for the auto-tester to finish.

A unittest is rarely a waste of everyone's time. `auto-merge` is here for a reason. Please don't systematically disregard the review comments you are getting.
